ui/src/app/app-routing.module.ts

```diff
@@ -36,343 +34,325 @@ const routes: Routes = [
   {
     path: 'user-disabled',
     component: AppRouting
+  },
+    // legacy / duplicated routes go HERE
+  {
+    path: '',
+    component: AppRouting,
+    data: {}
+  },
+  {
+    path: 'data-code-of-conduct',
+    component: AppRouting,
+    data: {}
+  },
+  {
+    path: 'nih-callback',
+    component: AppRouting,
+    data: {}
+  },
+  {
+    path: 'ras-callback',
+    component: AppRouting,
+    data: {}
+  },
+  {
+    path: 'access-renewal',
+    component: AppRouting,
+    data: {}
+  },
+  {
+    path: 'profile',
+    component: AppRouting,
+    data: {}
   },
   {
-    /*
-     * TODO angular2react: ideally, we should be able to just render the AppComponent here and let
-     *  React drive the rest of the routing but the BrowserRouter isn't able to pick up changes to the
-     *  route when it's embedded in Angular this way.. Until angular router is removed, we still need to
-     *  declare all of our routes in Angular which has the side effect of forcing a rerender of the
-     *  AppRouting component on every route change.
-     */
     path: '',
-    canActivate: [SignInGuard],
-    canActivateChild: [SignInGuard, DisabledGuard],
-    canDeactivate: [NavigationGuard],
-    runGuardsAndResolvers: 'always',
     children: [
       // legacy / duplicated routes go HERE
       {
-        path: '',
-        component: AppRouting,
-        data: {}
-      },
-      {
-        path: 'data-code-of-conduct',
-        component: AppRouting,
-        data: {}
-      },
-      {
-        path: 'nih-callback',
-        component: AppRouting,
-        data: {}
-      },
-      {
-        path: 'ras-callback',
-        component: AppRouting,
-        data: {}
-      },
-      {
-        path: 'access-renewal',
-        component: AppRouting,
-        data: {}
-      },
-      {
-        path: 'profile',
+        path: 'library',
         component: AppRouting,
         data: {}
       },
-      // non-migrated routes go HERE
       {
-        path: '',
-        runGuardsAndResolvers: 'always',
+        path: 'workspaces',
         children: [
           // legacy / duplicated routes go HERE
           {
-            path: 'library',
+            path: '',
             component: AppRouting,
             data: {}
           },
-          // non-migrated routes go HERE
           {
-            path: 'workspaces',
+            path: 'build',
+            component: AppRouting,
+            data: {}
+          },
+          {
+            /* TODO The children under ./views need refactoring to use the data
+             * provided by the route rather than double-requesting it.
+             */
+            path: ':ns/:wsid',
+            // TODO: When removing ng router, replace this with react-router's Prompt component.
```

Comment:
Could you please either create a ticket for tearing out the angular router or attach this TODO to RW-1920, which is likely where we'd have to tear this out anyways?

Reply:
Thanks, good call. This is in our style guide now after all.